api/v1/backend/BackendHandler.inc.php

```diff
@@ -0,0 +1,193 @@
+	 * 		@option int offset Default: 0
+	 * }
+	 *
+	 * @return Response
+	 */
+	public function getSubmissions($slimRequest, $response, $args) {
+
+		$request = $this->getRequest();
+		$currentUser = $request->getUser();
+		$context = $request->getContext();
+
+		// Merge query params over default params
+		$defaultParams = array(
+			'count' => 20,
+			'offset' => 0,
+		);
+
+		$params = array_merge($defaultParams, $slimRequest->getQueryParams());
+
+		// Anyone not a manager or site admin can only access their assigned
+		// submissions
+		if (!$currentUser->hasRole(array(ROLE_ID_MANAGER, ROLE_ID_SITE_ADMIN), $context->getId())) {
+			$defaultParams['assignedTo'] = $currentUser->getId();
+		}
+
+		// Process query params to format incoming data as needed
+		foreach ($params as $param => $val) {
+			switch ($param) {
+
+				// Always convert status to array
+				case 'status':
+					if (strpos($val, ',') > -1) {
+						$val = explode(',', $val);
+					} elseif (!is_array($val)) {
+						$val = array($val);
+					}
+					$params[$param] = array_map('intval', $val);
+					break;
+
+				case 'assignedTo':
+					$params[$param] = (int) $val;
+					break;
+
+				// Enforce a maximum count to prevent the API from crippling the
+				// server
+				case 'count':
+					$params[$param] = min(100, (int) $val);
+					break;
+
+				case 'offset':
+					$params[$param] = (int) $val;
+					break;
+
+				case 'orderBy':
+					if (!in_array($val, array('id', 'dateSubmitted', 'lastModified'))) {
+						unset($params[$param]);
+					}
+					break;
+
+				case 'orderDirection':
+					$params[$param] = $val === 'ASC' ? $val : 'DESC';
+					break;
+			}
+		}
+
+		// Prevent users from viewing submissions they're not assigned to,
+		// except for journal managers and admins.
+		if (!$currentUser->hasRole(array(ROLE_ID_MANAGER, ROLE_ID_SITE_ADMIN), $context->getId())
+				&& $params['assignedTo'] != $currentUser->getId()) {
+			return $response->withStatus(403)->withJsonError('api.submissions.403.requestedOthersUnpublishedSubmissions');
+		}
+
+		import('lib.pkp.classes.core.ServicesContainer');
+		$submissions = ServicesContainer::instance()
+				->get('submission')
+				->getSubmissionList($context->getId(), $params);
+
+		return $response->withJson($submissions);
+	}
+
+	/**
+	 * Delete a submission
+	 *
+	 * @param $slimRequest Request Slim request object
+	 * @param $response Response object
+	 * @param array $args arguments
+	 * @return Response
+	 */
+	public function deleteSubmission($slimRequest, $response, $args) {
+
+		$request = Application::getRequest();
+		$currentUser = $request->getUser();
+		$context = $request->getContext();
+
+		$submissionId = (int) $args['submissionId'];
+
+		$submissionDao = Application::getSubmissionDAO();
+		$submission = $submissionDao->getById($submissionId);
+
+		if (!$submission) {
+			return $response->withStatus(404)->withJsonError('api.submissions.404.resourceNotFound');
```

Comment:
does withJsonError function exists in Slim 3? I have been using the syntax below for error.
`$response->withJson(array('error' => 'api.submissions.404.resourceNotFound'), 404);`